cve_bin_tool/cvedb.py

```diff
@@ -501,6 +438,11 @@ def populate_purl2cpe(self):
         cve_conn = sqlite3.connect(self.dbpath)
         cve_cursor = cve_conn.cursor()
 
+        # we are occasionally seeing an error where the cache doesn't have
+        # purl2cpe and thus we get an error, so attempt to initalize here
+        cve_cursor.execute(self.TABLE_SCHEMAS["purl2cpe"])
+        cve_cursor.execute(self.INDEXES["purl"])
+
```

Comment:
I proposed a change in commit [622efe8](https://github.com/intel/cve-bin-tool/pull/4236/commits/622efe82455caf4a1b2c2dd47e1e2c8197ff1ab9). With this change, we should no longer face this issue. So, it's not necessary, but I don't know we can have it as a added security?

Reply:
Let's try your change out first and see how it goes!    As in, I'll plan to merge your script before this PR gets completed.

I think I still want the rest of the refactoring in here, though, so I'll continue to work to get that behaving rather than closing this PR.  I'll decide later if we need to take out these two lines or (since I don't think they'll add much overhead) if they should just say in as a backup safety check.